apps/wallet-mobile/src/features/Scan/common/useTriggerScanAction.tsx

```diff
@@ -0,0 +1,56 @@
+import {useSend} from '../../../features/Send/common/SendContext'
+import {useSelectedWallet} from '../../../SelectedWallet/Context/SelectedWalletContext'
+import {pastedFormatter} from '../../../yoroi-wallets/utils/amountUtils'
+import {asQuantity, Quantities} from '../../../yoroi-wallets/utils/utils'
+import {ScanAction, ScanFeature} from './types'
+import {useNavigateTo} from './useNavigateTo'
+
+export const useTriggerScanAction = ({insideFeature}: {insideFeature: ScanFeature}) => {
+  const {receiverChanged, amountChanged, tokenSelectedChanged, resetForm, memoChanged} = useSend()
+  const {primaryTokenInfo} = useSelectedWallet()
+  const navigateTo = useNavigateTo()
+
+  const trigger = (scanAction: ScanAction) => {
+    switch (scanAction.action) {
+      case 'send-single-pt': {
+        navigateTo.back()
+        navigateTo.send()
+
+        if (insideFeature !== 'send') resetForm()
```

Comment:
when launching the feature from another must reset the state (e.g reading a `stake` link within the send funnel)